#1449 - Integration of the PRACC / HPRM Manager

kvanhijf opened the pull request.

kvanhijf committed:
#1449 - Integration of the PRACC / HPRM Manager

khenderick commented on ovs/lib/storagerouter.py:
```diff
@@ -1264,6 +1265,128 @@ def get_logfiles(local_storagerouter_guid):
+        These configuration will be zipped and made available for download
+        :param storagerouter_guid: The guid of the StorageRouter for which a HPRM manager needs to be deployed
+        :type storagerouter_guid: str
+        :param local_storagerouter_guid: The guid of the StorageRouter this call is executed on
```
It's not about this call, but where the API was executed

khenderick commented on ovs/lib/storagerouter.py:
```diff
@@ -1264,6 +1265,128 @@ def get_logfiles(local_storagerouter_guid):
+        fragment_cache_on_write = parameters['fragment_cache_on_write']
+        if fragment_cache_on_read is True or fragment_cache_on_write is True:
+            if 'path' in parameters:  # Local caching
+                required_params.update({'path': (str, re.compile('^/$|/.*[^.]$')),
```
That doesn't make much sense to me, are you sure this works? I'd expect something like `^/.*[^.]?$`

khenderick commented on ovs/lib/storagerouter.py:
```diff
@@ -1264,6 +1265,128 @@ def get_logfiles(local_storagerouter_guid):
+                Toolbox.verify_required_params(actual_params=parameters,
+                                               required_params=required_params)
+                alba_backend_name = parameters['backend_info']['alba_backend_name']
+                if Configuration.dir_exists(key='/ovs/arakoon/{0}-abm'.format(alba_backend_name)) is False:
```
If it's an external cluster, you'll have to use the API calls like we use in the `add_vpool` logic.

khenderick commented on ovs/lib/storagerouter.py:
```diff
@@ -1264,6 +1265,128 @@ def get_logfiles(local_storagerouter_guid):
+        file_contents_map = {'/opt/OpenvStorage/config/{0}/config.json'.format(vpool.name): config,
+                             '/opt/OpenvStorage/config/{0}/arakoon.ini'.format(vpool.name): Configuration.get(key=config_path.format('abm'), raw=True)}
+        if fragment_cache_info[0] == 'alba':
+            file_contents_map['/opt/OpenvStorage/config/{0}/accelerated_arakoon.ini'.format(vpool.name)] = Configuration.get(key='/ovs/arakoon/{0}-abm/config'.format(alba_backend_name), raw=True)
```
This should then be the actual config as loaded from the API.

khenderick commented on webapps/api/backend/views/storagerouters.py:
```diff
@@ -153,6 +153,44 @@ def get_support_info(self, storagerouter):
+        :param vpool_guid: Guid of the vPool for which the proxy is configured
+        :type vpool_guid: str
+        """
+        return StorageRouterController.get_proxy_config.s(vpool_guid=vpool_guid,
```
Isn't `.delay(...)` easier?

khenderick commented on webapps/frontend/app/viewmodels/wizards/createhprmconfigs/data.js:
```diff
@@ -0,0 +1,120 @@
+    "use strict";
+    var pathRegex, hostRegex, singleton, parsePresets;
+
+    pathRegex = /^\/$|\/.*[^.]$/;
```
Same remark as about the regex above.

khenderick commented on webapps/frontend/app/viewmodels/wizards/createhprmconfigs/gather.js:
```diff
@@ -0,0 +1,292 @@
+        };
+        self.reloadBackends = function() {
+            self.reloadingBackends(true);
+            $.when.apply($, [self.loadBackends()])
```
This is a bit strange.

kvanhijf committed:
#1449 - ArakoonClusterConfig now has a convert format method to go from INI <--> JSON
Removed regex for local path, instead using simple checks in BLL
UseLocalBackend flag correctly configured when relevant in the 'Create HPRM config files' wizard

kvanhijf committed:
#1449 - Unittest for ArakoonClusterConfig.convert_format

khenderick committed:
Merge branch 'develop' into pracc-integration

khenderick committed:
Integration of fragment and block caches + Various fixes

JeffreyDevloo commented on ovs/lib/storagerouter.py:
```diff
@@ -1396,6 +1396,156 @@ def get_logfiles(local_storagerouter_guid):
+    def create_hprm_config_files(storagerouter_guid, local_storagerouter_guid, parameters):
+        """
+        Create the required configuration files to be able to make use of HPRM (aka PRACC)
+        These configuration will be zipped and made available for download
```
This* configuration

JeffreyDevloo commented on ovs/lib/storagerouter.py:
```diff
@@ -1396,6 +1396,156 @@ def get_logfiles(local_storagerouter_guid):
+
+        cache_info = {}
+        arakoons = {}
+        cache_types = ['fragment', 'block']
```
Could we define these somewhere in an enum? I figure it has been used on some more places.

JeffreyDevloo commented on ovs/lib/storagerouter.py:
```diff
@@ -1396,6 +1396,156 @@ def get_logfiles(local_storagerouter_guid):
+                                                          'size': (int, {'min': 1, 'max': 10 * 1024})})
+                        Toolbox.verify_required_params(actual_params=parameters,
+                                                       required_params=required_params)
+                        while '//' in path:
```
No need to iterate here, by default replace will replace all occurences

khenderick replied:
This small loop will also replace `/////` for example.

JeffreyDevloo commented on ovs/lib/storagerouter.py:
```diff
@@ -1396,6 +1396,156 @@ def get_logfiles(local_storagerouter_guid):
+                        path = path.strip()
+                        if not path or path.endswith('/.') or '..' in path or '/./' in path:
+                            raise ValueError('Invalid path specified')
+                        required_params[ctype][1].update({'path': (str, None),
```
Define an argument for referring to this dict to up the readability.

JeffreyDevloo commented on ovs/lib/storagerouter.py:
```diff
@@ -1396,6 +1396,156 @@ def get_logfiles(local_storagerouter_guid):
+                                               required_params=required_params)
+                read = parameters[ctype]['read']
+                write = parameters[ctype]['write']
+                if read is True or write is True:
```
if read is False and write is False:
 cache_info[ctype] = ['none']
 continue
INSERT THE CURRENT IF CODE HERE

JeffreyDevloo commented on ovs/lib/storagerouter.py:
```diff
@@ -1396,6 +1396,156 @@ def get_logfiles(local_storagerouter_guid):
+        for file_name, contents in file_contents_map.iteritems():
+            local_client.file_write(contents=contents,
+                                    filename=file_name)
+        local_client.run(command='tar --transform "s#^config/{0}#{0}#" -czf /opt/OpenvStorage/webapps/frontend/downloads/{1} config/{0}'.format(identifier, tgz_name), allow_insecure=True)
```
Insecure necessary here?

khenderick committed:
Small fixes based on review
